frontend/src/metabase/dashboard/actions/save.js

```diff
@@ -159,8 +159,7 @@ export const saveDashboardAndCards = createThunkAction(
 
       await dispatch(Dashboards.actions.update(dashboard));
 
-      // make sure that we've fully cleared out any dirty state from editing (this is overkill, but simple)
-      dispatch(fetchDashboard(dashboard.id, null)); // disable using query parameters when saving
+      dispatch(fetchDashboard(dashboard.id, window.location.search, true));
```

Comment:
Just to be on the safe side, let's try to preserve the previous behavior for dashboards? I mean:

```js
if (dashboard.is_app_page) {
  dispatch(fetchDashboard(dashboard.id, window.location.search, true));
} else {
  // make sure that we've fully cleared out any dirty state from editing (this is overkill, but simple)
  dispatch(fetchDashboard(dashboard.id, null)); // disable using query parameters when saving
}

Reply:
Yeah good point. Down to hedge. 👍